### ISSUE TITLE
Translate pages are taking a long time to load

### ISSUE THREAD

**Issue by muedli** [phetsims/rosetta#236]
Figure out why loading pages takes so long.

When you hit the "Go" button on the `/translate/` page, the next page takes a really long time to load / maybe never loads?

**Comment by muedli** [comment]
I did a quick investigation of this. I pulled on my macOS laptop and ran `npm run dev`. Then I tried to get to `/translate/sim/acid-base-solutions/ab`. Initially, this was taking a long time. The info and debug statements in the console stopped at "Retrieving any previously-saved-but-not-submitted strings." This statement is from routeHandlers.js.

```
  // Create a parameterized query string for retrieving the user's previously saved strings.
  const savedStringsQuery = 'SELECT * FROM saved_translations WHERE user_id = $1 AND locale = $2 AND (' + repositories + ')';

  // Connect to the database and query for saved strings corresponding to this user and sim.
  let rows = null;
  try {
    winston.info( `Retrieving any previously-saved-but-not-submitted strings. Query: ${savedStringsQuery}.` );
    try {

      // Execute the query using the template query string and parameters for user ID and target locale.
      const queryResponse = await pool.query( savedStringsQuery, [ userId, targetLocale ] );
      rows = queryResponse.rows;
      winston.info( 'Retrieval of previously-saved-but-not-submitted strings succeeded.' );
    }
    catch( error ) {
      winston.error( `Retrieval of previously-saved-but-not-submitted strings failed. Error: ${error}.` );
    }
  }
  catch( error ) {
    winston.error( `Unable to retrieve saved strings from database. Error: ${error}.` );
  }
```

This led me to look at the changes I made to routeHandlers.js. I looked through the commits and I didn't see anything obvious. (Most of the changes I made were cosmetic.)

Thus, I decided to revert my local copy of Rosetta to an old commit and and work my way forward chronologically to see where the slow loading was introduced. I started at a583c260dfab340b0f1d63b4d1f9a36115a2ca08, and `/translate/sim/acid-base-solutions/ab` loaded in a reasonable time frame. I then worked my way through the commits, checking to see how long `/translate/sim/acid-base-solutions/ab` took to load for each commit. For some reason, they all loaded in a reasonable time frame. Master is loading in a reasonable time frame for me now. I tried loading the page in a different browser to see if it was a caching issue, but that wasn't the issue. I'm now unable to reproduce the original slow loading of the `/translate/sim/acid-base-solutions/ab` page.

**Comment by muedli** [comment]
The above investigation leads me to think the issue might have crept in with a8aea03456ec339804aa41a5b8fc13b49b79e279 where I upgraded the "pg" module from 7.14.0 to 8.2.1. The "pg" module is used for interacting with the short-term storage database.

Then again, I'm not sure I'm barking up the right tree. Maybe the issue doesn't have to do with routeHandlers.js or the upgrade of the "pg" module. I need to figure out how to reproduce the issue again.

**Comment by muedli** [comment]
Remember to `npm prune` and `npm update` each time you check a commit!

**Comment by muedli** [comment]
Strange new development: this is no longer happening for me. I checked on my Windows machine and my Mac machine. Then grabbed the latest version of Rosetta from master on phet-server-dev, and I didn't see any slow loading on ox-dev.colorado.edu.

1. Check with John to see if this is still happening on his machine.
2. Make sure the process I used to update Rosetta on phet-server-dev was correct.

As for 2, I did the following:
* SSH'd to phet-server-dev.
* Went to the Rosetta repo and ran `git pull`.
* Restarted Rosetta and checked the logs to verify that I had the newest version. (I changed something in the logs recently, so this was an easy way to check.)
* I then went to ox-dev.colorado.edu/translate to test the changes.

**Comment by muedli** [comment]
@jbphet and I determined that the slow loading we were seeing was caused by not connecting to a short-term storage database in a reasonable time frame. It's possible that when the pg module was upgraded, something changed with how connecting to the short-term database works, and this is why the slow loading went away. In https://github.com/brianc/node-postgres/blob/master/CHANGELOG.md, the maintainers of the pg module note performance improvements for pg@8.2.0, so maybe that's where the changes came in.

Next item on the docket is making sure that if you don't have a short-term storage database set up, the connection times out after two seconds.
